packages/grid/x-data-grid-pro/src/components/headerFiltering/GridHeaderFilterItem.tsx

```diff
@@ -0,0 +1,279 @@
+            break;
+
+          default:
+            if (
+              event.metaKey ||
+              event.ctrlKey ||
+              event.altKey ||
+              event.shiftKey ||
+              event.key === 'Tab'
+            ) {
+              break;
+            }
+            apiRef.current.startHeaderFilterEditMode(colDef.field);
+        }
+      },
+      [apiRef, colDef.field, colIndex, columnFields, headerFilterMenuRef, isEditing, isMenuOpen],
+    );
+
+    const publish = React.useCallback(
+      (eventName: keyof GridHeaderFilterEventLookup, propHandler?: React.EventHandler<any>) =>
+        (event: React.SyntheticEvent) => {
+          apiRef.current.publishEvent(
+            eventName,
+            apiRef.current.getColumnHeaderParams(colDef.field),
+            event as any,
+          );
+          if (propHandler) {
+            propHandler(event);
+          }
+        },
+      [apiRef, colDef.field],
+    );
+
+    const onClick = React.useCallback(
+      (event: React.MouseEvent) => {
+        if (!hasFocus) {
+          apiRef.current.startHeaderFilterEditMode(colDef.field);
+          apiRef.current.setColumnHeaderFilterFocus(colDef.field, event);
+        }
+      },
+      [apiRef, colDef.field, hasFocus],
+    );
+
+    const mouseEventsHandlers = React.useMemo(
+      () => ({
+        onKeyDown: publish('headerFilterKeyDown', onKeyDown),
+        onClick: publish('headerFilterClick', onClick),
+      }),
+      [onClick, onKeyDown, publish],
+    );
+
+    const ownerState = {
+      ...rootProps,
+      colDef,
+    };
+
+    const classes = useUtilityClasses(ownerState as OwnerState);
+
+    const isNoInputOperator = NO_INPUT_OPERATORS[colDef.type!]?.includes(item.operator);
+    const isFilterActive = hasFocus || Boolean(item?.value) || isNoInputOperator;
+
+    const placeholder = apiRef.current.getLocaleText('columnMenuFilter');
+
+    return (
+      <div
+        className={clsx(classes.root, headerClassName)}
+        ref={handleRef}
+        style={{
+          height,
+          width,
+          minWidth: width,
+          maxWidth: width,
+        }}
+        role="columnheader"
+        aria-colindex={colIndex + 1}
+        aria-label={headerFilterComponent == null ? colDef.headerName ?? colDef.field : undefined}
+        {...other}
+        {...mouseEventsHandlers}
+      >
+        {headerFilterComponent}
+        {InputComponent && !headerFilterComponent ? (
+          <InputComponent
+            apiRef={apiRef}
+            item={item}
+            inputRef={inputRef}
+            applyValue={applyFilterChanges}
+            onFocus={() => {
+              apiRef.current.startHeaderFilterEditMode(colDef.field);
+            }}
+            onBlur={() => {
+              apiRef.current.stopHeaderFilterEditMode();
+            }}
+            fullWidth
+            placeholder={placeholder}
+            label={isFilterActive ? OPERATOR_LABEL_MAPPING[item.operator] : ' '}
```

Comment:
We need to think about making `OPERATOR_LABEL_MAPPING` localizable. I know that you haven't marked the localization as done but we can see if it woundn;t be too difficult to add it to the first version.

Reply:
That's a good point, I am actually thinking to update the labels in the original filter panel too, i.e wherever we have symbols instead of values, that way we can open them up for localization and make this consistent across filter panels.

Example of such labels:
https://github.com/mui/mui-x/blob/10162e2b973d93fd3b110e33ed83652cf5e3542b/packages/grid/x-data-grid/src/colDef/gridNumericOperators.ts#L30-L31

The `label` could be `equals` instead and then we can map it with the respective localizations.

Reply:
Added localization for all the operators except numeric, which I believe to be a breaking change as we have to change `operator.value` for those operators which have symbols used as value (=, !=, >, >=, < and <=)
We can cater this in v7 probably.